moralization/transformers_model_manager.py

```diff
@@ -310,12 +318,134 @@ def load_optimizer(self, learning_rate: float = 2e-5, kwargs: dict = None) -> No
+        true_labels = [
+            [self.label_names[m] for m in label if m != -100] for label in labels
+        ]
+        true_predictions = [
+            [self.label_names[p] for (p, m) in zip(prediction, label) if m != -100]
+            for prediction, label in zip(predictions, labels)
+        ]
+        return true_labels, true_predictions
+
+    def train(self) -> None:
+        """Train a model using the pre-loaded components."""
+
+        # show a progress bar
+        progress_bar = tqdm(range(self.num_training_steps))
+
+        for epoch in range(self.num_train_epochs):
+            # set the mode to training
+            self.model.train()
+            for batch in self.train_dataloader:
+                outputs = self.model(**batch)
+                loss = outputs.loss
+                self.accelerator.backward(loss)
+                self.optimizer.step()
+                self.lr_scheduler.step()
+                self.optimizer.zero_grad()
+                progress_bar.update(1)
+
+            # Evaluation
+            self._evaluate_model()
+
+            self.results = self.metric.compute()
+            print(
+                f"epoch {epoch}:",
+                {
+                    key: self.results[f"overall_{key}"]
+                    for key in ["precision", "recall", "f1", "accuracy"]
+                },
+            )
+
+            self.save()
+
+    def _evaluate_model(self):
+        # set mode to evaluation
+        self.model.eval()
+        for batch in self.eval_dataloader:
+            with no_grad():
+                outputs = self.model(**batch)
+            predictions = outputs.logits.argmax(dim=-1)
+            labels = batch["labels"]
+            # Necessary to pad predictions and labels for being gathered
+            predictions = self.accelerator.pad_across_processes(
+                predictions, dim=1, pad_index=-100
+            )
+            labels = self.accelerator.pad_across_processes(
+                labels, dim=1, pad_index=-100
+            )
+            predictions_gathered = self.accelerator.gather(predictions)
+            labels_gathered = self.accelerator.gather(labels)
+            true_predictions, true_labels = self.postprocess(
+                predictions_gathered, labels_gathered
+            )
+            self.metric.add_batch(predictions=true_predictions, references=true_labels)
+
+    def save(self):
+        # Save the model to model path
+        self.accelerator.wait_for_everyone()
+        unwrapped_model = self.accelerator.unwrap_model(self.model)
+        unwrapped_model.save_pretrained(
+            self._model_path, save_function=self.accelerator.save
+        )
+        # the below is executed only once in main process
+        if self.accelerator.is_main_process:
+            self.tokenizer.save_pretrained(self._model_path)
+
+    def evaluate(self, token: str, model_path: Union[str, Path] = None):
```

Comment:
I find it surprising I can pass a `model_path` here, since the TransformersModelManager object has `self._model_path` - I would find it more natural to instantiate a new manager to evaluate a different model:
```
tmm2 = TransformersModelManager("my_other_model_path")
tmm2.evaluate("blah blah")
```